students/JerryH/Lesson05/mailroom3.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python3
+from collections import defaultdict
+
+# List of sample donors and donations
+donors = defaultdict(list, {
+    "Bill Gates": [234.22, 45645.24, 43953.09, 98823],
+    'Jeff Bezo': [4564.23],
+    'Mike Dell': [299.09, 26273.67],
+    'Harry Potter': [8234.09, 83948.04, 7834.23],
+    'Ben Williams': [29283.00, 1334.34],
+    'Guy James': [93.00, 34.34]
+    })
+
+menu = {
+    'op1': "Send a Thank You",
+    'op2': "Create a Report",
+    'op3': "Send Letters To Everyone",
+    'op4': "Quit"
+    }
+
+QUIT_OPT = '4'
+
+def prompt():
+    return input("Please choose the following options:\n1) {op1}.\n2) {op2}.\n3)"
+        " {op3}.\n4) {op4}\n".format(**menu))
+
+
+def donor_name_prompt():
+    return input("Send a Thank You - Please enter a full name or type \"list\""
+        "to list the current donors:\n")
+
+
+def get_all_donor_names():
+    return list(donors)
+    
+
+def donation_prompt():
+    return float(input("Please enter the donation amount:\n"))
+
+
+def list_all_donor_names():
+    for each_name in sorted(get_all_donor_names()):
+        print("{}".format(each_name))
+
+
+def send_thank_you():
+    donor_name = None
+    while not donor_name:
+        donor_name = donor_name_prompt()
+        if donor_name.lower() == "list":
+            donor_name = None
+            list_all_donor_names()
+
+    donation = None
+    while not donation:
+        try:
+            donation = donation_prompt()
+        except:
```

Comment:
because of the exception `donation` will not have a value assigned so no need to set `donation = None`